packages/server/src/middlewares/validate.ts

```diff
@@ -1,12 +1,27 @@
 import { Context } from "koa";
 import { accountSchema } from "../validation";
-export const validateCreateAccountBody = (ctx: Context, next: () => void) => {
+import { getAccountByEmail } from "../db/accounts";
+export const validateCreateAccountData = async (
+  ctx: Context,
+  next: () => void,
+) => {
   const { error } = accountSchema.validate(ctx.request.body);
 
   if (error) {
     ctx.status = 403;
-    ctx.body = "Missing parameters";
+    ctx.message = "Missing parameters";
     return;
   }
+
+  const { email } = ctx.request.body;
+
+  const alreadyExist = !!(await getAccountByEmail(email));
+
+  if (alreadyExist) {
+    ctx.status = 401;
```

Comment:
```suggestion
    ctx.status = 409;
```
409 stands for `Conflict` => https://en.wikipedia.org/wiki/List_of_HTTP_status_codes